ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/objectstore/JournalStoreTest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright The Narayana Authors
+ *
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+package com.hp.mwtests.ts.arjuna.objectstore;
+
+import com.arjuna.ats.arjuna.common.ObjectStoreEnvironmentBean;
+import com.arjuna.ats.arjuna.common.Uid;
+import com.arjuna.ats.arjuna.exceptions.ObjectStoreException;
+import com.arjuna.ats.arjuna.objectstore.StoreManager;
+import com.arjuna.ats.arjuna.state.OutputObjectState;
+import com.arjuna.ats.internal.arjuna.objectstore.hornetq.HornetqJournalEnvironmentBean;
+import com.arjuna.ats.internal.arjuna.objectstore.hornetq.HornetqObjectStoreAdaptor;
+import com.arjuna.common.internal.util.propertyservice.BeanPopulator;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+public class JournalStoreTest {
+    private static HornetqJournalEnvironmentBean hornetqJournalEnvironmentBean;
+    private final String typeName = "/StateManager/junit";
+
+    @BeforeClass
+    public static void before() {
+        BeanPopulator.getDefaultInstance(ObjectStoreEnvironmentBean.class).
+                setObjectStoreType(HornetqObjectStoreAdaptor.class.getName());
+
+        hornetqJournalEnvironmentBean = BeanPopulator.getDefaultInstance(HornetqJournalEnvironmentBean.class);
+    }
+
+    @AfterClass
+    public static void after() {
+        hornetqJournalEnvironmentBean.setMaxIO(2);
+        hornetqJournalEnvironmentBean.setAsyncIO(false);
+    }
+
+    @Test
+    public void test1() throws ObjectStoreException {
```

Comment:
it would be good to name the test method based on its functionality than with an abstract name like `test1`

Reply:
The tests a extremely simple and easier to parse than would using a name that reflects what the tests does:

```
testMaxIOEqualsOneWithAIO
testMaxIOEqualsTwoWithAIO
testMaxIOEqualsOneWithNIO
testMaxIOEqualsTwoWithNIO
```

It is harder to read those than what my PR proposes.

Reply:
I like the `testMaxIOEqualsTwoWithAIO` more but as it's matter of taste. It could be as it is, I'm fine.

Reply:
That would mean retesting the PR which would shorten the window that we have to get this into the 5.11 branch, tested on our CI, released, and then the tasks that would need doing to get the workaround into EAP.

Reply:
@ochaloup So when we get some resolution on this issue I will backport it and also https://issues.redhat.com/browse/JBTM-3496 to the 5.11 branch. Then we can tag and do a release of the 5.11 branch.

Reply:
I mean I'm fine with the original name. I pointed it as a minor objection and despite that I've already approved the PR. If you consider your naming better let's merge it as it is.

Reply:
I will merge then (although I will have no objection if someone renames them sometime in the future.